NoSuchElementException occurred when create mask rule.

```
[ERROR] 2023-04-08 18:18:28.767 [ShardingSphere-Command-1] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.util.NoSuchElementException: null
	at java.base/java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:721)
	at java.base/java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:741)
	at org.apache.shardingsphere.infra.metadata.database.schema.util.SchemaMetaDataUtils.lambda$addOneActualTableDataNode$3(SchemaMetaDataUtils.java:99)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.apache.shardingsphere.infra.metadata.database.schema.util.SchemaMetaDataUtils.addOneActualTableDataNode(SchemaMetaDataUtils.java:99)
	at org.apache.shardingsphere.infra.metadata.database.schema.util.SchemaMetaDataUtils.getSchemaMetaDataLoaderMaterials(SchemaMetaDataUtils.java:63)
	at org.apache.shardingsphere.infra.metadata.database.schema.builder.GenericSchemaBuilder.loadSchemas(GenericSchemaBuilder.java:98)
	at org.apache.shardingsphere.infra.metadata.database.schema.builder.GenericSchemaBuilder.build(GenericSchemaBuilder.java:76)
	at org.apache.shardingsphere.infra.metadata.database.schema.builder.GenericSchemaBuilder.build(GenericSchemaBuilder.java:64)
	at org.apache.shardingsphere.infra.metadata.database.ShardingSphereDatabase.create(ShardingSphereDatabase.java:89)
	at org.apache.shardingsphere.metadata.factory.ExternalMetaDataFactory.create(ExternalMetaDataFactory.java:50)
	at org.apache.shardingsphere.metadata.MetaDataFactory.create(MetaDataFactory.java:48)
	at org.apache.shardingsphere.mode.manager.ContextManager.createChangedDatabases(ContextManager.java:399)
	at org.apache.shardingsphere.mode.manager.ContextManager.createMetaDataContexts(ContextManager.java:362)
	at org.apache.shardingsphere.mode.manager.ContextManager.alterRuleConfiguration(ContextManager.java:252)
	at org.apache.shardingsphere.mode.manager.standalone.StandaloneModeContextManager.alterRuleConfiguration(StandaloneModeContextManager.java:255)
	at org.apache.shardingsphere.proxy.backend.handler.distsql.rdl.rule.RuleDefinitionBackendHandler.execute(RuleDefinitionBackendHandler.java:100)
	at org.apache.shardingsphere.proxy.backend.handler.distsql.rdl.rule.RuleDefinitionBackendHandler.execute(RuleDefinitionBackendHandler.java:75)
	at org.apache.shardingsphere.proxy.backend.handler.distsql.rdl.RDLBackendHandler.execute(RDLBackendHandler.java:98)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:92)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.doExecuteCommand(CommandExecutorTask.java:114)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:109)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:77)
	at com.alibaba.ttl.TtlRunnable.run(TtlRunnable.java:60)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
CREATE DATABASE TEST;

USE TEST;

CREATE MASK RULE t_mask (
COLUMNS(
(NAME=phone_number,TYPE(NAME='MASK_FROM_X_TO_Y', PROPERTIES("from-x"=1, "to-y"=2, "replace-char"="*"))),
(NAME=address,TYPE(NAME='MD5'))
));
```

--- a/infra/common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/util/SchemaMetaDataUtils.java
+++ b/infra/common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/util/SchemaMetaDataUtils.java
@@ -96,6 +96,9 @@ public final class SchemaMetaDataUtils {
     private static void addOneActualTableDataNode(final GenericSchemaBuilderMaterial material,
                                                   final Map<String, Collection<String>> dataSourceTableGroups, final DataNodes dataNodes, final String table) {
         Optional<DataNode> dataNode = dataNodes.getDataNodes(table).stream().filter(each -> isSameDataSourceNameSchemaName(material, each)).findFirst();
+        if (!dataNode.isPresent() && !material.getDataSourceMap().keySet().iterator().hasNext()) {
+            return;
+        }
         String dataSourceName = dataNode.map(DataNode::getDataSourceName).orElseGet(() -> material.getDataSourceMap().keySet().iterator().next());
         String tableName = dataNode.map(DataNode::getTableName).orElse(table);
         addDataSourceTableGroups(dataSourceName, tableName, dataSourceTableGroups);

